web/models/setting/system_variable.ex

```diff
@@ -79,6 +73,26 @@ defmodule CoursePlanner.SystemVariable do
   end
   defp validate_value_type(changeset), do: changeset
 
+  def parse_value(value, type) do
+    case type do
+      "string"  -> {:ok, value}
+      "list"     -> parse_list(value)
+      "integer" -> parse_integer(value)
+      "boolean" -> parse_boolean(value)
+      _         -> {:error, "unknown type"}
+    end
+  end
+
+  def parse_list(value) do
+    parse_list =
+    value
```

Comment:
I am not sure it is more understandable, but you may replace these 4 lines by
```elixir
String.split(value, ~r/,\s*/, trim: true)
```
what do you think?